samples/Jzon.cpp

```diff
@@ -843,83 +843,65 @@ namespace Jzon
 		std::string valueBuffer;
 		bool saveBuffer;
 
-		for (; cursor < jsonSize; ++cursor)
```

Comment:
I did not check what this `cursor` is doing, but it looks like a private member variable that was increased in those loops, and not it is not increased anymore. So conditions such as `cursor == jsonSize - 1` might be broken now.

We would need to check quietly if this variable and the conditions around it are important.

Reply:
I think we should leave cursor in the code.   The code was written by a guy in Sweden.  We can contact him, if you feel this requires attention.

I think cursor is the read position in a string and it's set to zero in the class Parse constructor.

I've put a statement `if ( cursor) std::out << "cursor = " << cursor << std::endl;` before the `for` statement and it isn't executed on Stonehenge.jpg.

I've tried removing it from the class and making it a local variable.  That can't be done.  There are member functions inspecting cursor.

```c++
	char Parser::peek()
	{
		if (cursor < jsonSize-1)
		{
			return json.at(cursor+1);
		}
		else
		{
			return '\0';
		}
	}
	void Parser::jumpToNext(char c)
	{
		++cursor;
		while (cursor < jsonSize && json.at(cursor) != c)
			++cursor;
	}
	void Parser::jumpToCommentEnd()
	{
		++cursor;
        char c1 = '\0';
		for (; cursor < jsonSize; ++cursor)
		{
            char c2 = json.at(cursor);

			if (c1 == '*' && c2 == '/')
				break;

			c1 = c2;
		}
	}
```